src/mono/mono/mini/debug-mini.c

```diff
@@ -78,7 +79,8 @@ mono_debug_open_method (MonoCompile *cfg)
 	g_assert (header);
 
 	info->jit = jit = g_new0 (MonoDebugMethodJitInfo, 1);
-	info->line_numbers = g_array_new (FALSE, TRUE, sizeof (MonoDebugLineNumberEntry));
+	info->line_numbers = dn_vector_alloc_t (MonoDebugLineNumberEntry);
+	dn_checkfail (info->line_numbers != NULL, "Allocation failed");
```

Comment:
Thinking a little about this pattern, could an alternative be to have a new vector attribute that we can set when doing custom alloc/init that we fail fast internal, like DN_VECTOR_ATTRIBUTE_FAIL_FAST. That code path is already in the slow path so check this attribute and fail fast if requested instead of returning NULL should not add overhead to fast path. It will be an opt in and you need to use the custom init/alloc to get it, but will reduce need to check returns due to allocation failures, that could happen to a number of different functions, alloc/init/push_back/resize etc.

Reply:
We could also add a couple of pre defined init structs that enable some values, so instead of needing to declare a struct with this flag set, you can just use the predeclared struct in your custom init/alloc call.

Reply:
Honestly that all sounds like a huge pain.

I would prefer something like

```
  dn_vector_t *vec = DN_AOK (dn_vector_alloc_t (...));
```

with something like

```c
#define DN_AOK(expr) ({ void *_ck = (expr); dn_checkfail(_ck != NULL, "Allocation failed"); _ck})
```

except:
1. with proper macros so the file:line is from the caller of `DN_AOK()`, not from the  macro definiton
2. with some `typeof` trick so the whole expression has the same type as `expr`
3. without using the GCC `({ stmt; expr })` non-standard extension

Reply:
I think it could be rather straightforward, and it would simplify case to work similar to how g_array works where it asserts on internal allocation failures, so you could make sure a dn_vector_t would behave the same way with regards to internal allocation failures.

The macro way works as well, but they are not exclusive, both variations could exist. Handle it using macros will of course generate more code around each call to a vector function that could fail (that we didn't check for failures when using g_array). The macro solution would probably be able to validate both pointer as well as bool return cases for failures.